parlai/tasks/ccpe/agents.py

```diff
@@ -3,150 +3,106 @@
-                    ):
-                        entry = []
-                        entry.append(userCnt)
-                        entry.append(currUtt['text'])
-                        entry.append([flattenedData[ep][i - 1]['text']])
-                        entry.append(currUtt['segments'])
-                        entry.append(flattenedData[ep][i - 1]['segments'])
-                        entry.append(False)
-                        currUserEp.append(entry)
-                        userCnt += 1
-                    if (
-                        currUtt['speaker'] == 'ASSISTANT'
-                        and flattenedData[ep][i - 1]['speaker'] == 'USER'
-                    ):
-                        entry = []
-                        entry.append(asssistantCnt)
-                        entry.append(currUtt['text'])
-                        entry.append([flattenedData[ep][i - 1]['text']])
-                        entry.append(currUtt['segments'])
-                        entry.append(flattenedData[ep][i - 1]['segments'])
-                        entry.append(False)
-                        currAssistantEp.append(entry)
-                        asssistantCnt += 1
-
-            currUserEp[-1][5] = True
-            currAssistantEp[-1][5] = True
-            self.userData.append(currUserEp)
-            self.assistantData.append(currAssistantEp)
-
-        self.data = self.assistantData + self.userData
-
-    def get(self, episode_idx, entry_idx=0):
-        ep = self.data[episode_idx]
-        entry = ep[entry_idx]
-        action = {
-            'id': entry[0],
-            'text': entry[1],
-            'labels': entry[2],
-            'textSegments': entry[3],
-            'labelSegments': entry[4],
-            'episode_done': entry[5],
-        }
-        return action
-
-    def share(self):
-        shared = super().share()
-        shared['data'] = self.data
-        return shared
-
-
-class CCPEAssistantTeacher(CCPEAllTeacher):
-    def _setup_data(self):
-        super()._setup_data()
-        self.data = self.assistantData
-
-
-class DefaultTeacher(CCPEAllTeacher):
+        # do a 80:10:10 train/valid/test split
+        full_episode_count = len(json_data) // 10
+        if fold == 'train':
+            json_data = json_data[: full_episode_count * 8]
+        elif fold == 'valid':
+            json_data = json_data[full_episode_count * 8 : full_episode_count * 9]
+        elif fold == 'test':
+            json_data = json_data[full_episode_count * 9 :]
+
+        episodes = []
+        for conversation in json_data:
+            episode = []  # Assume all turns are alternating with ASSISTANT first
+            idx = 0
+            utts = conversation["utterances"]
+            if utts[0]["speaker"] != "ASSISTANT":
+                # Make a dummy assistant turn and add a user turn
+                episode.append({"speaker": "ASSISTANT", "text": "", "segments": []})
```

Comment:
usually we'll do something like put a `__SILENCE__` token here; has your model handled these empty text observations well?

Reply:
Putting in a `__SILENCE__` token seems like a good call; hadn't thought of that.